internal/security/secretstore/init.go

```diff
@@ -74,7 +78,7 @@ func NewBootstrap(insecureSkipVerify bool, vaultInterval int) *Bootstrap {
 	return &Bootstrap{
 		insecureSkipVerify: insecureSkipVerify,
 		vaultInterval:      vaultInterval,
-		validKnownSecrets:  map[string]bool{redisSecretName: true, messagebusSecretName: true},
+		validKnownSecrets:  map[string]bool{redisSecretName: true, postgresSecretName: true, messagebusSecretName: true},
 	}
```

Comment:
check the database type from the config first, and add valid known secret for either redisSecretName or postgresSecretName

Reply:
Updated to only add `postgresSecretName` if database type is postgres.